src/controllers/subscription.controller.ts

```diff
@@ -0,0 +1,20 @@
+import { Request, Response } from 'express';
+import { validationResult } from 'express-validator';
+import { StatusCode } from '../models/status-codes.model';
+import { SubscriptionsService } from '../services/subscription.service';
+
+export async function subscribe(request: Request, response: Response) {
+  const subscriptionService = new SubscriptionsService();
+  const errors = validationResult(request);
+  if (!errors.isEmpty()) {
+    return response.status(StatusCode.UnprocessableEntity).json(errors.array());
```

Comment:
The use of 422 or 400 error codes is an open question. The 422 error is rarely used. 400 is when there is no required field, the data type is incorrect. 422 is when the value of `email` field is not an email address. In this case, validation checks everything at once, whether the field is present, whether the field is a string, and whether it is an email. In the first two checks, it is more correct to use 400, in the last 422. But more often they do not bother with this and just answer 400.
You will rarely see anyone supporting the 422 code.

This is a comment to think about 🤔
Does not block MR

Reply:
Agreed, and it seems that more general error code would be better for this case. Updated